### ISSUE TITLE
Seperate Go-routine for Jobs

### ISSUE THREAD

**Issue by haseth** [robfig/cron#291]
I wanted to discuss another method of running cron jobs:

Currently, we are storing all the ```Jobs``` in Entries and we sort them based on ```byTime``` function. We run the first job. 
So, in ```Run``` method we do evaluation of all the ```Entries```  and sorting... 

Why haven't we run each Job in separate go-routine? 
- With timer 
- and Stop channel 

Wouldn't it be a better implementation? I might be wrong in the approach but I have raised a prototype code for it, which might give better clarity what I am trying to say. 

- I could add a new job easily by invoking another go-routine.
- Remove Job could be done by sending signal to ```Entry Stop Channel```. 
- Stop would stop all the jobs by sending ```stop``` to all the go routines
- Snapshot of all the entries could be taken.

**Comment by robfig** [comment]
I agree, I think that approach could work well for job scheduling in general. 

To be honest, I didn't even consider implementing it like that. I implemented similarly to the Vixie cron, and I found the resulting code (at least v1 of it) to be pretty simple, so I never felt the need to introduce more goroutines. I suppose one benefit of the current model is that it's a bit less resource intensive, since it doesn't use resources for idle goroutines.

Thinking about it, your proposal does have one limitation though: it wouldn't be able to run multiple instances of a job concurrently. I'm not sure how important that is, but the contract of a cron is to start each job on the given schedule, and that would be impossible with this design without that caveat (that jobs are skipped if a previous instance is still running).
